### ISSUE TITLE
 ICU Lexing Error: Unexpected character

### ISSUE THREAD

**Issue by awaik** [flutter/flutter#119802]
After upgrading Flutter I have a lot of errors:

1.
```
 [app_sk.arb:settings__homepage_tip] ICU Lexing Error: Unexpected character.
    Zobrazuje sa vo vašom verejnom profile. Prijíma sa iba platná webová adresa (napr. https://{example_url})
                                                                                                                                                     ^
```

2.
```
[app_sw.arb:agoradesk250Sbguide250Sbsearch250Sbchoose8722Sbad8722Sbtype] ICU Lexing Error: Unexpected character.
    Katika safu wima ya kushoto unapaswa kuchagua kichupo cha '{buyOrSellTabName}}'. Ikiwa ungependa kubadilisha nchi, sarafu, njia ya kulipa au kubainisha kiasi kinachohitajika, bofya kitufe cha 'Onyesha menyu ya utafutaji', chagua vigezo unavyotaka, na ubonyeze kitufe cha kutafuta.
                                                                                  ^
Exception: Found syntax errors.
```

And many others.

Can you please advise me, on how can I understand what to fix?

<details>
  <summary>Doctror</summary>

```
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.1, on macOS 13.0.1 22A400 darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
[✓] VS Code (version 1.74.1)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!

```
```

**Comment by huycozy** [comment]
@awaik 
Can you share a complete string in `arb` file? 

It seems the character `{`causes the issue. Please check the solution at https://github.com/flutter/flutter/issues/119375#issuecomment-1410986306 that may help to solve the issue.

**Comment by awaik** [comment]
I guess, there is a luck of migration docs - for example, now is not possible to use
1. `{example_url}`
2. `{example-url}`
3. ....

I fixed it with the script, but made a lot of iterations to figure out what to change.

**Comment by huycozy** [comment]
@awaik Does that solution work for you? The documentation for this will be added later as mentioned at [comment](https://github.com/flutter/flutter/issues/119375#issuecomment-1410986306).

**Comment by thkim1011** [comment]
By the way the documentation has been updated to include how to escape parts of the syntax. See here: https://docs.flutter.dev/development/accessibility-and-localization/internationalization#escaping-syntax.

**Comment by rscircus** [comment]
Just ran also into this... hm, the fixes described in #120098 don't seem to work for me. Neither for tag `v2.9.0` nor for `v2.9.2`...
